Update truffle and tests to error on Revert

maurelian opened the pull request:
I upgraded truffle to 4.0.6 to get the proper `revert` message. 

This is a pre-req for integrating solidity-coverage.

maurelian commented on migrations/2_deploy_tokens.js:
```diff
@@ -1,5 +1,5 @@
 
 module.exports = (deployer) => {
-  deployer.deploy(EIP20);
+  deployer.deploy(EIP20, 10000, 'Simon Bucks', 1, 'SBX');
```
The truffle upgrade was erroring without constructor arguments.

maurelian commented on test/eip20/eip20.js:
```diff
@@ -1,4 +1,4 @@
-const { expectThrow } = require('../utils');
+const { assertRevert } = require('../utils');
```
This is similar to Simon's 721 WIP.

maurelian commented on test/utils.js:
```diff
@@ -1,12 +1,11 @@
-    }
-    assert.fail(errMsg);
-  },
+  assertRevert: async promise => {
```
This is similar to Simon's 721 WIP, so merging that should be a bit less painful.

skmgoldin commented on test/utils.js:
```diff
@@ -1,12 +1,11 @@
+  assertRevert: async promise => {
+  try {
+    await promise
+    assert.fail('Expected revert not received')
```
This `assert.fail` should go _after_ the catch block, shouldn't it? Otherwise we'll catch this error, and it does indeed contain the "revert" substring. So put this after (outside) the catch block, and put a return at the end of (inside) the catch block.

maurelian replied:
I think after the catch block is equivalent. If `promise` returns as an error, we go immediately to the catch block. If `promise` is successful, then it passes to `assert.fail()`.

skmgoldin replied:
Does `assert.fail` not emit an error like `assert(false)`?

skmgoldin replied:
Looking at the API, I think this will work as-written, but I think it's not using the API properly. I think it should be `assert.fail(foo, bar, 'Expected revert not received')`, and if it _were_ that, it would hit the catch with a `revert` substring in the error message.

skmgoldin replied:
See additional discussion here: https://consensys.slack.com/archives/C037XE8SU/p1519666129000572

skmgoldin commented:
Also, I know you hate staging, but would you mind making this PR to staging?

maurelian commented:
@skmgoldin FYI, I've learned to stop worrying and love the staging. 

Other changes:
* fixed conflicts with staging
* improved linting scripts, which were missing some files (FYI, `eslint --debug` is a good way to check which files get linted)
* implemented `assertRevert` just the way you like it.

skmgoldin commented:
LFGTM
